packages/modules/devices/huawei/huawei/bat.py

```diff
@@ -11,24 +11,29 @@
 from modules.common.simcount import SimCounter
 from modules.common.store import get_bat_value_store
 from modules.devices.huawei.huawei.config import HuaweiBatSetup
+from modules.devices.huawei.huawei.type import HuaweiType
 
 
 class HuaweiBat(AbstractBat):
     def __init__(self,
                  device_id: int,
                  component_config: Union[Dict, HuaweiBatSetup],
-                 modbus_id: int) -> None:
+                 modbus_id: int,
+                 type: HuaweiType) -> None:
         self.__device_id = device_id
         self.component_config = dataclass_from_dict(HuaweiBatSetup, component_config)
         self.modbus_id = modbus_id
+        self.type = type
         self.sim_counter = SimCounter(self.__device_id, self.component_config.id, prefix="speicher")
         self.store = get_bat_value_store(self.component_config.id)
         self.fault_state = FaultState(ComponentInfo.from_component_config(self.component_config))
 
     def update(self, client: ModbusTcpClient_) -> None:
-        time.sleep(1)
+        if type == HuaweiType.SDongle:
```

Comment:
```suggestion
        if self.type == HuaweiType.SDongle:
```
Entsprechend auch an den anderen Stellen. Sonst greift er auf eine Python-Klasse namens `type` zu.